nextest-runner/src/config.rs

```diff
@@ -345,4 +414,74 @@ mod tests {
             .profile(NextestConfig::DEFAULT_PROFILE)
             .expect("default profile should exist");
     }
+
+    #[test_case(
+        "",
+        SlowTimeout { period: Duration::from_secs(60), terminate_after: Some(0)},
+        None
+
+        ; "empty config is expected to use the hardcoded values"
+    )]
+    #[test_case(
+        indoc! {r#"
+            [profile.default]
+            slow-timeout = "30s"
+        "#},
+        SlowTimeout { period: Duration::from_secs(30), terminate_after: None },
+        None
+
+        ; "overrides the default profile"
+    )]
+    #[test_case(
+        indoc! {r#"
+            [profile.default]
+            slow-timeout = "30s"
+
+            [profile.ci]
+            slow-timeout = { period = "60s", terminate-after = 3 }
+        "#},
+        SlowTimeout { period: Duration::from_secs(30), terminate_after: None, },
+        Some(SlowTimeout { period: Duration::from_secs(60), terminate_after: Some(3), })
+
+        ; "adds a custom profile 'ci'"
```

Comment:
Could you add more test cases here:

```toml
[profile.default]
slow-timeout = { period = "60s" }
```

```toml
[profile.default]
slow-timeout = { period = "60s", terminate-after = 0 }
```

```toml
[profile.default]
slow-timeout = { period = "60s", terminate-after = 3 }

[profile.ci]
slow-timeout = "30s"
```

```toml
[profile.default]
slow-timeout = "60s"

[profile.ci]
slow-timeout = { terminate-after = 3 }
```

This one I think should error out -- if you're specifying `terminate-after` you should also specify the period to make it really clear how long you're going to terminate after.